ui/guest/src/elementRegistry.ts

```diff
@@ -154,11 +156,13 @@ export function register(payload: ElementRecordRegistration): number {
     // for the model to be loaded.
     if (isInheritedField(model.craftercms.id, fieldId)) {
       byPathFetchIfNotLoaded(model.craftercms.sourceMap?.[fieldId]).subscribe((response) => {
+        const modelIdToRegister = response.craftercms.id;
+        inheritorsModelIdsMap[`${modelIdToRegister}-${fieldId}`] = modelId;
```

Comment:
Let's create some sort of util to create that key
```suggestion
        fieldHeirsMap[createFieldHeirsKey(modelIdToRegister, fieldId)] = modelId;
```